src/main/java/meteordevelopment/meteorclient/utils/entity/ProjectileEntitySimulator.java

```diff
@@ -114,21 +114,41 @@ public void set(Entity user, double roll, double speed, double simulated, double
         this.waterDrag = waterDrag;
     }
 
-    public boolean set(Entity entity, boolean accurate, double tickDelta) {
+    public boolean set(Entity entity, boolean accurate, double tickDelta, boolean noGravity) {
         // skip entities in ground
         if (entity instanceof PersistentProjectileEntity && ((ProjectileInGroundAccessor) entity).getInGround()) return false;
 
         if (entity instanceof ArrowEntity arrow) {
             // im not sure if arrow.getVelocity().length() is correct but it works ¯\_(ツ)_/¯
-            set(entity, arrow.getVelocity().length(), 0.05000000074505806, 0.6, accurate, tickDelta);
+            double gravity = 0.05000000074505806;
+            if (noGravity) {
```

Comment:
Instead of adding this to every case you can just overwrite the gravity before `return true;` at the end of the method.